### PR TITLE
V3.1.6

### PR DESCRIPTION
**Pull request by leelalagudu**
*Issue #, if available:* #43

*Description of changes:*

- Fixes exception handling by using the SDK provided named exceptions
- Refactors logging constructs to allow config provided log level setting for core and current config import lambda functions

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by leelalagudu** [comment]
@jmejco , @tamara-h  , the security check project would fail as the CI build spec needs to be updated to reflect "FunctionlogMode" key in config.

File: config/leela.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+---
+App: "aws-sso-extensions-for-enterprise"
+Environment: "leela"
```
Comment (jmejco) [inline config/leela.yaml:3]: Please remove this file.

Comment (leelalagudu) [reply config/leela.yaml]: Pushed with the changes

File: config/env.yaml [hunk 2]
```diff
@@ -28,3 +28,4 @@ Parameters:
   ImportCurrentSSOConfiguration: false # Set this to true if you want the solution to do a one-time import of your current AWS SSO permission sets and account assignments. Refer to "Import existing AWS SSO access entitlements for management through the solution" section under "Features" in README.md for details
   UpgradeFromVersionLessThanV303: false # Should be set to true, if upgrading from solution version 3.0.2/earlier
   SupportNestedOU: false # Set this to true if your preference is for the solution to support nested OU's as part of account assignments. Refer to "Enterprise friendly account assignment life cycle" and "Automated access change management for root, ou_id and account_tag scopes" sections under "Features" in README.md for details
+  FunctionlogMode: "Info" # Used for configuring lambda function logging level. Alowed values - ["Debug","Info","Warn","Exception"]
```
Comment (jmejco) [inline config/env.yaml:31]: Shouldn't it be `FunctionLogMode` ?

Comment (leelalagudu) [reply config/env.yaml]: Corrected this now

**Comment by jmejco** [comment]
@leelalagudu Code LGTM. Will deploy and run tests before approving.

**Review by jmejco — APPROVED** [review]

**Pull request merged**